src/tree/mod.rs

```diff
@@ -88,6 +81,16 @@ impl RatchetTree {
         }
     }
 
+    /// Create a new `RatchetTree` by cloning the public tree nodes from another tree
+    /// and an empty `PrivateTree`
+    pub(crate) fn new_from_public_tree_values(ratchet_tree: &RatchetTree) -> Self {
```

Comment:
Maybe `from_tree` or something? We pass in the entire tree, not only public values.
But this should really be implemented in the `Clone` trait of `RatchetTree` I guess, as that's what this is really.

Reply:
I renamed it. We discussed that ratchet trees should not be clonable and that's why I implemented it like this. Of course I can also implement `Clone` again, but it might be misleading, because you might assume that all values from the tree are then cloned.